DepositRegistrar.sol

```diff
@@ -0,0 +1,306 @@
+    */ 
+    function newBid(bytes32 sealedBid) {
+        if (sealedBids[sealedBid] > 0 ) throw;
+        // creates a new Name contract with the owner
+        address newBid = new Deed(msg.sender, sealedBid, this);
+        sealedBids[sealedBid] = newBid;
+        if (!newBid.send(msg.value)) throw;
+    } 
+    
+    
+    /*
+    ## Winning bids are deposited
+
+    The highest bid gets control of the name, by depositing the equivalent amount of the second 
+    highest bid. The funds will be held on new contract, controlled only by the owner which 
+    contains basic information like when the name was first registered, when it needs to be 
+    renewed, and the price paid for it. **The price paid is also saved on a moving average amount, 
+    calculated as ```averagePrice = averagePrice * 0.999 + newPrice * 0.001```**. The 
+    averagePrice at the moment of purchase is also registered on the contract.
+    */ 
+    function revealBid(bytes32 _hashedName, address _owner, uint _value, bytes32 _salt) noEther  {
+        bytes32 seal = shaBid(_hashedName, _owner, _value, _salt);
+        address bidAddress = sealedBids[seal];
+        if (bidAddress == 0 ) throw;
+        Deed bid = Deed(bidAddress);
+        bid.unsealBid(_hashedName, _owner, _value, _salt);
+        entry name = hashedNames[bid.hashedName()];
+        
+        if (bid.creationDate() > name.auctionDate - revealPeriod
+            || now > name.auctionDate ) {
+            // bid is invalid, burn 99.9%
+            bid.closeDeed(999);
+            
+        } else if ( _value < minBid ) {
+            // bid is invalid but not punishable, burn 0.1%
+            bid.closeDeed(1);
+            
+        } else if (bid.value() > name.highestBid) {
+            // new winner
+            // cancel the other bid, burn 0.1%
+            Deed previousWinner = Deed(name.deed);
+            previousWinner.closeDeed(1);
+            
+            // set new winner
+            name.value = name.highestBid;
+            name.highestBid = bid.value();
+            name.deed = sealedBids[seal];
+        
+        } else if (bid.value() > name.value) {
+            // not winner, but affects second place
+            name.value = bid.value();
+            bid.closeDeed(1);
+            
+        } else {
+            // bid doesn't affect auction
+            bid.closeDeed(1);
+        }
+    }
+    
+    function finalizeAuction(bytes32 nameHash) noEther {
+        entry name = hashedNames[nameHash];
+        if (now < name.auctionDate || name.status != Mode.Auction) throw;
+        
+        // set the name
+        name.status = Mode.Owned;
+        name.firstRegistered = now;
+        name.lastRenewed = now;
+
+        //set the averages
+        uint period = (now - lastSinceNewName) * M;
+        averagePeriod = (9999 * averagePeriod + period) / 10000;
+        uint n = averagePeriod < (60 seconds * M) ? 7 days * M / averagePeriod : 10000;
```

Comment:
It checks if there were, in average more than 10,000 registration per week (averagePeriod less than 60.48 seconds). If so, use the number of average registrations, elsewhere, use 10,000. I'm using period as a simpler way to measure frequency.